### DISCUSSION TITLE
Compiling SCSS/CSS is slow, but quick in node modules?

### DISCUSSION THREAD

**Discussion by laugepetersen** [webpack/webpack#14891] [Q&A]
The confusing.

When compiling my scss files from my main entry project the compile and bundle time is around 3000-5000 ms.
<img width="583" alt="billede" src="https://user-images.githubusercontent.com/24874516/144483699-03bd607d-8250-4458-b455-6e564af845bc.png">
<img width="408" alt="billede" src="https://user-images.githubusercontent.com/24874516/144483778-85a8c400-b6a3-4184-a253-427d3074532a.png">

BUT! When i compile a scss file within node_modules it only takes 50-100 ms? How come?
<img width="583" alt="billede" src="https://user-images.githubusercontent.com/24874516/144483945-51e1d0b3-80cf-4b40-9863-df85435c88fa.png">
<img width="345" alt="billede" src="https://user-images.githubusercontent.com/24874516/144484008-20df30ba-04df-4e66-9f7c-aeba881b7b13.png">

I do use cache-loader and it helped a lot, but still wondering how this has an effect changing a scss files from my project directory versus a scss files within foundation-sites from the node_modules directory for an instance.

**Comment by alexander-akait** [comment]
You can't change files in `node_modules`, they are cached and modification doesn't change their content

**Reply by laugepetersen** [reply]
All right, makes sense. But any suggestions on quickening the SCSS bundling so it becomes as fast as the javascript bundling?

**Reply by alexander-akait** [reply]
Maybe you have big SCSS file, what is size of SCSS file?

**Reply by laugepetersen** [reply]
Can't be that big. All i have yet is foundations SCSS files such as settings which import all its necessary files. Nothing else yet.
<img width="628" alt="billede" src="https://user-images.githubusercontent.com/24874516/144487518-d93174f8-5f8f-4abc-b6b3-cc99bca63708.png">

**Reply by laugepetersen** [reply]
i have used gulp for a long times and it does the css handling a lot better and faster. But i decided to switch to Webpack since all its other advantages. It seems like i need the best of both worlds or some solution to webpack.

**Reply by alexander-akait** [reply]
What is your configuration for CSS pipeline (i.e. can you describe loaders for CSS)?

**Reply by laugepetersen** [reply]
The cache-loader did almost no performance enchancing but the Fiber plugin did A LOT. But yet still 3000-5000 ms.

<img width="717" alt="billede" src="https://user-images.githubusercontent.com/24874516/144488090-44fc12de-b935-47a3-88b0-2664132c5420.png">

**Reply by alexander-akait** [reply]
Anyway, please remove `cache-loader`, you don't need it, webpack v5 supports cache out of box https://webpack.js.org/configuration/cache/#cachetype

Also what is postcss plugins? Maybe you don't need to run all of them (for example autoprefixer) for development builds

Also can you show `magicImporter`? buit-in importer should work good

**Reply by laugepetersen** [reply]
I didn't knew what magicImporter did, but it was suggested somewhere. I can outcomment.

PostCSS from webpack:
<img width="773" alt="billede" src="https://user-images.githubusercontent.com/24874516/144488805-157dd0ca-86e4-4ff1-9e91-14b84f3a2f41.png">

PostCss.config.js
<img width="786" alt="billede" src="https://user-images.githubusercontent.com/24874516/144488878-bb88ac73-485b-4217-90b8-30f884aab396.png">

**Reply by alexander-akait** [reply]
Disable `purge-css` and `autoprefixer` for envelopment builds, most likely they take up most of the time

**Reply by laugepetersen** [reply]
Okay, thanks will try to do so and make an update here.

Also something i notice is the sass-dart errors that come up everytime on save and bundling, coming from foundation library.-

I think that slows it too.

**Reply by alexander-akait** [reply]
Yes, errors and warnings output reduce time, there are options to disable them

**Reply by alexander-akait** [reply]
Also try to disable custom importer for scss-loader

**Reply by laugepetersen** [reply]
> Yes, errors and warnings output reduce time, there are options to disable them

I tried finding a solution, but it seemed to be impossible to get away from them? How do i go about it then?


> Also try to disable custom importer for scss-loader

Where is this in my code? Not sure i know what you mean

**Reply by alexander-akait** [reply]
> I tried finding a solution, but it seemed to be impossible to get away from them? How do i go about it then?

Why it is impossible?

> Where is this in my code? Not sure i know what you mean

`importer`

**Reply by laugepetersen** [reply]
I just read many people saying that you can't get rid of sass-darts warnings.

**Reply by alexander-akait** [reply]
What are warnings?

**Reply by laugepetersen** [reply]
Removing postCss loader and its settings such as autoprefixer helped. now it seems build time around 2000ms solid.

But, the warnings i would like to get away. Here is an example

<img width="574" alt="billede" src="https://user-images.githubusercontent.com/24874516/144599713-eb466b50-0eab-40ea-8716-9c8431052503.png">

**Reply by alexander-akait** [reply]
Try `quietDeps` https://github.com/sass/sass/issues/3065
